parquet_file/src/catalog.rs

```diff
@@ -0,0 +1,1237 @@
+        path.set_file_name(format!("{}.{}", Uuid::new_v4(), TRANSACTION_FILE_SUFFIX));
+        create_empty_file(&object_store, &path).await;
+
+        // file is folder
+        let mut path = transactions_path(&object_store, server_id, &db_name);
+        path.push_dir("0");
+        path.push_dir(format!("{}.{}", Uuid::new_v4(), TRANSACTION_FILE_SUFFIX));
+        path.set_file_name("foo");
+        create_empty_file(&object_store, &path).await;
+
+        // top-level file
+        let mut path = transactions_path(&object_store, server_id, &db_name);
+        path.set_file_name(format!("{}.{}", Uuid::new_v4(), TRANSACTION_FILE_SUFFIX));
+        create_empty_file(&object_store, &path).await;
+
+        // no data present
+        let option = PreservedCatalog::load(Arc::clone(&object_store), server_id, db_name.clone())
+            .await
+            .unwrap();
+        assert!(option.is_none());
+
+        // can still write + read
+        assert_catalog_roundtrip_works(&object_store, server_id, &db_name).await;
+    }
+
+    #[tokio::test]
+    async fn test_missing_transaction() {
+        let object_store = make_object_store();
+        let server_id = make_server_id();
+        let db_name = "db1";
+        let trace = assert_single_catalog_inmem_works(&object_store, server_id, db_name).await;
+
+        // remove transaction file
+        assert!(trace.tkeys.len() >= 2);
+        let tkey = &trace.tkeys[0];
+        let path = transaction_path(&object_store, server_id, db_name, tkey);
+        checked_delete(&object_store, &path).await;
+
+        // loading catalog should fail now
+        let res =
+            PreservedCatalog::load(Arc::clone(&object_store), server_id, db_name.to_string()).await;
+        assert!(matches!(
+            res,
+            Err(Error::MissingTransaction {
+                revision_counter: 0
+            })
+        ));
+    }
+
+    #[tokio::test]
+    async fn test_transaction_version_mismatch() {
+        let object_store = make_object_store();
+        let server_id = make_server_id();
+        let db_name = "db1";
+        let trace = assert_single_catalog_inmem_works(&object_store, server_id, db_name).await;
+
+        // break transaction file
+        assert!(trace.tkeys.len() >= 2);
+        let tkey = &trace.tkeys[0];
+        let path = transaction_path(&object_store, server_id, db_name, tkey);
+        let mut proto = load_transaction_proto(&object_store, &path).await.unwrap();
+        proto.version = 42;
+        store_transaction_proto(&object_store, &path, &proto)
+            .await
+            .unwrap();
+
+        // loading catalog should fail now
+        let res =
+            PreservedCatalog::load(Arc::clone(&object_store), server_id, db_name.to_string()).await;
+        assert!(res.is_err());
+        assert_eq!(
+            res.err().unwrap().to_string(),
+            "Format version of transaction file for revision 0 is 42 but only [1] are supported"
+        );
```

Comment:
```suggestion
        assert_eq!(
            res.unwrap_err().unwrap().to_string(),
            "Format version of transaction file for revision 0 is 42 but only [1] are supported"
        );
```

May be slightly less verbose.